fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/FcrepoClientFactory.java

```diff
@@ -24,11 +24,24 @@
  */
 public class FcrepoClientFactory {
 	
-	private static String baseUri;
+	private String baseUri;
 	
-	public FcrepoClientFactory(String base) {
+	private static FcrepoClientFactory instance;
+	
+	private FcrepoClientFactory(String base) {
 		baseUri = base;
 	}
+	
+	public static FcrepoClientFactory factory(String baseUri) {
+		if (instance == null) {
+			instance = new FcrepoClientFactory(baseUri);
+		}
+		return instance;
+	}
+
+	public static FcrepoClientFactory defaultFactory() {
```

Comment:
I was having a hard time figuring out how to best handle that situation.  In this implementation, when you create a FcrepoClientFactory and provide a baseUri it stores the constructed factory as the singleton instance of that factory.  You can then call FcrepoClientFactory.defaultFactory() elsewhere in the code and not have to provide a baseUri since it is just returning the originally constructed copy.  But it requires FcrepoClientFactory.factory(baseUri) to have been called first, otherwise .defaultFactory would return null, which isn't so great.